### PR TITLE
Use httpx as http client library

### PR DESCRIPTION
**Pull request by pletnes**

### CODE REVIEW

**Comment by pletnes** [comment]
@robertofalk this patch makes cloud-pysec use [httpx](https://www.python-httpx.org/) as its http client. All unit tests pass, but I have not yet had the chance to test this with our applications that use cloud-pysec.

I have noted one strange behavior though: the tests (on master) actually perform real network requests to `api.cf.test.com`, which seems strange - `test.com` is registered to a company in Florida - do you know the reason for this?

Note that this PR also contains the changes done in #52, so I would suggest performing that merge first.

**Comment by robertofalk** [comment]
Tests failed, anyway I'll check the `api.cf.test.com` situation in the meantime.

File: tests/test_req_token.py [hunk 1]
```diff
@@ -39,8 +42,9 @@ def setUpClass(cls):
         environ["SAP_EXT_JWT_ALG"] = "*"
 
         cls.DEVNULL = open(devnull, 'w')
-        cls.flask_process = Popen(['flask', 'run', '-p', flask_port, '-h', 'localhost'],
-                                  env=flask_env, stdout=cls.DEVNULL, stderr=cls.DEVNULL)
+        cls.flask_process = Popen(['/sapmnt/home/I506278/git/github/sap/cloud-pysec/.venv/bin/flask',
```
Comment (robertofalk) [inline tests/test_req_token.py:45]: tests are failing because of this I believe

Comment (pletnes) [reply tests/test_req_token.py]: Yes, I added that in order to enable debugging in my editor. The `flask` executable is not on PATH, it's just  sitting inside a virtualenv for me. I've fixed it now.

We could start the "mock app" in-process, avoiding the need for `Popen` - just a thought.

**Comment by pletnes** [comment]
@robertofalk I've added python3.9 for the testing and now all tests pass, also on travis, not just my machine 😃 I take it you want to test the library running under at least one of your applications before merging to master?

**Review by pletnes — COMMENTED** [review]
.

**Comment by pletnes** [comment]
@robertofalk I managed to use this httpx-only branch with my application, and it worked successfully, as far as I can tell. Did you have a chance to try it out?

**Review by robertofalk — APPROVED** [review]

**Comment by robertofalk** [comment]
Hi @pletnes, we are still testing it but in the basic tests that I already ran, everything is fine. So in order to test in a real application, I need a release, therefore it's released.

**Pull request merged**